File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.kt
class Solution {
    fun solution(s: String): IntArray {
        var str = s
        var transformCnt = 0
        var removedZeroCnt = 0

        while (str != "1") {
            val len = str.length
            val ones = str.count { it == '1' }

            removedZeroCnt += (len - ones)
            transformCnt++

            // 다음 문자열 = 1의 개수의 2진수 표현
            str = Integer.toBinaryString(ones)
        }

        return intArrayOf(transformCnt, removedZeroCnt)
    }
}

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.js
function solution(s) {
    let transformCnt = 0;
    let removedZeroCnt = 0;

    while (s !== "1") {
        const len = s.length;
        let ones = 0;

        for (let i = 0; i < len; i++) {
            if (s[i] === '1') ones++;
        }

        removedZeroCnt += (len - ones);
        transformCnt++;

        // 다음 문자열 = 1의 개수의 2진수 표현
        s = ones.toString(2);
    }

    return [transformCnt, removedZeroCnt];
}